### PR TITLE
HOCS-1930-Msg-Pdf-Converter

### PR DESCRIPTION
**Pull request by Paulcr-ho**
This upgrade of the document converter includes the ability to convert MSG (Email Message file) with image attachments.

The following updates are included:
1. the auxilii MsgParser and iTextPdf tool libs
2. the MSG Document converter, for Email file type conversion with image attachments (tif. tiff, jpg, jpeg, gif, png)
3. msg sample files for testing
4. unit tests for functionality verification

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/document/MSGDocumentConverter.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+        final byte[] outputBytes = arrayOutputStream.toByteArray();
+        arrayOutputStream.close();
+        return outputBytes;
+    }
+
+    private MsgContents extractContents(Message message) {
+        MsgContents contents = new MsgContents();
+        contents.setFromEmail(message.getFromEmail());
+        contents.setFromEmail(contents.getFromEmail().contains("Content_Types") ? StringUtils.EMPTY : contents.getFromName());
```
Comment (hubert-raich) [inline src/main/java/uk/gov/digital/ho/hocs/document/MSGDocumentConverter.java:111]: two lines (111 and 112 both setting the same property).

File: src/main/java/uk/gov/digital/ho/hocs/document/MSGDocumentConverter.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        } catch (RuntimeWorkerException | DocumentException | IOException e) {
+            return false;
+        }
+        return true;
+    }
+
+    private void processAttachment(Document pdf, Attachment attachment) {
+        // we don't process messages containing messages
+        if (!(attachment instanceof FileAttachment)) {
+            return;
```
Comment (hubert-raich) [inline src/main/java/uk/gov/digital/ho/hocs/document/MSGDocumentConverter.java:141]: Scattered returns within methods are hard to follow, it would be nicer if we changed this if statement and included the logic for FileAttachment within it.

File: src/main/java/uk/gov/digital/ho/hocs/document/MsgContents.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import com.auxilii.msgparser.attachment.Attachment;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+
+@Getter
+@Setter
+public class MsgContents {
```
Comment (hubert-raich) [inline src/main/java/uk/gov/digital/ho/hocs/document/MsgContents.java:12]: Class with no proper constructor but with setters for all its private variables.

I think we should use all args constructor, or even better a constructor that takes in Message directly:
MsgContents(Message message){
...
}

File: src/main/java/uk/gov/digital/ho/hocs/document/DocumentConversionResource.java [hunk 4]
```diff
@@ -66,6 +69,23 @@ public void convert(@RequestParam("file") MultipartFile file, HttpServletRespons
                 return;
             }
         }
+        if (msgDocumentConverter.isSupported(fileExtension)) {
```
Comment (hubert-raich) [inline src/main/java/uk/gov/digital/ho/hocs/document/DocumentConversionResource.java:72]: convert() method seem to be growing a lot, not a must have for this review but we should think about how to break up this method or make it shorter.

**Review by hubert-raich — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/digital/ho/hocs/document/DocumentConversionResource.java (hunk 4 above)
Comment (hubert-raich) [reply src/main/java/uk/gov/digital/ho/hocs/document/DocumentConversionResource.java:72]: this one is optional

**Comment by hubert-raich** [comment]
Few small changes, it is still a very good and neat solution for what we though would require its own windows based docker service.  Very good solution

File: src/main/java/uk/gov/digital/ho/hocs/document/MSGDocumentConverter.java (hunk 1 above)
Comment (Paulcr-ho) [reply src/main/java/uk/gov/digital/ho/hocs/document/MSGDocumentConverter.java:111]: Read it more carefully Hubert.

File: src/main/java/uk/gov/digital/ho/hocs/document/MSGDocumentConverter.java (hunk 2 above)
Comment (Paulcr-ho) [reply src/main/java/uk/gov/digital/ho/hocs/document/MSGDocumentConverter.java:141]: Personal preference has no place in a review.

File: src/main/java/uk/gov/digital/ho/hocs/document/MsgContents.java (hunk 3 above)
Comment (Paulcr-ho) [reply src/main/java/uk/gov/digital/ho/hocs/document/MsgContents.java:12]: Constructor is not required as it fields are set separately.
